lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -603,6 +609,168 @@ def build_module(self, trans, payload={}):
+            prediction = np.reshape(prediction, (prediction.shape[1],))
+
+            # normalize the predicted scores with max and sort the predictions
+            max_prediction = float(np.max(prediction))
+            if max_prediction == 0.0:
+                max_prediction = 1.0
+            prediction = prediction / max_prediction
+            prediction_pos = np.argsort(prediction, axis=-1)
+
+            # get topk prediction
+            topk_prediction_pos = prediction_pos[-topk:]
+            last_compatible_tools = self.compatible_tools[last_tool_name].split(",")
+
+            # read tool names using reverse dictionary
+            pred_tool_ids = [self.reverse_dictionary[int(tool_pos)] for tool_pos in topk_prediction_pos]
+            predicted_scores = [int(prediction[pos] * 100) for pos in topk_prediction_pos]
+
+            pred_tool_ids = list()
+            for tool_pos in topk_prediction_pos:
+                tool_name = self.reverse_dictionary[int(tool_pos)]
+                pred_tool_ids.append(tool_name)
+
+            pred_tool_ids_rev = list(reversed(pred_tool_ids))
+            predicted_scores_rev = list(reversed(predicted_scores))
+
+            # get the predicted tools
+            for child, score in zip(pred_tool_ids_rev, predicted_scores_rev):
+                c_dict = dict()
+                for t_id in self.all_tools:
+                    # select the name and tool id if it is installed in Galaxy
+                    if t_id == child and score > 0.0 and child in last_compatible_tools:
+                        full_tool_id = self.all_tools[t_id][0]
+                        pred_input_extensions, _ = self.__get_tool_extensions(trans, full_tool_id)
+                        c_dict["name"] = self.all_tools[t_id][1] + " (" + str(score) + "%)"
+                        c_dict["tool_id"] = full_tool_id
+                        c_dict["i_extensions"] = list(set(pred_input_extensions))
+                        prediction_data["children"].append(c_dict)
+                        break
+
+            # show only a few
+            prediction_data["children"] = prediction_data["children"][:to_show - 1]
+            # get a list of recommended tools set by the admin
+            if self.enable_admin_tool_recommendations is True or self.enable_admin_tool_recommendations == "true":
+                for item in self.admin_recommendations_list["tools"]:
+                    if last_tool_name == item["tool_id"]:
+                        prediction_data["children"].extend(item["recommendations"])
+                        break
+
+            # get the root name for displaying after tool run
+            for t_id in self.all_tools:
+                if t_id == last_tool_name:
+                    prediction_data["name"] = self.all_tools[t_id][1]
+                    break
+            return prediction_data
+        except Exception as exp:
+            return prediction_data
```

Comment:
..and it is not handled. So, essentially, it anything bad happens in this large block of code, it will fail silently and simply return the value. Instead, I think we should narrow down the `try/except`, log the *specific* error and either proceed or fail, depending on what that error is (and be explicit about it).